cmd/check_container.go

```diff
@@ -62,218 +63,214 @@ var checkContainerCmd = &cobra.Command{
-			if certProject.Container.Privileged {
-				cfg.EnabledChecks = engine.RootExceptionContainerPolicy()
-			}
-		}
+// checkContainer runs the Container policy.
+func checkContainer(ctx context.Context, cfg *runtime.Config) error {
+	cfg.Policy = policy.PolicyContainer
 
-		engine, err := engine.NewForConfig(cfg)
-		if err != nil {
-			return err
-		}
+	// configure the artifacts directory if the user requested a different directory.
+	if cfg.Artifacts != "" {
+		artifacts.SetDir(cfg.Artifacts)
+	}
 
-		formatter, err := formatters.NewForConfig(cfg)
+	// Determine if we need to modify the policy that's executed.
+	if cfg.CertificationProjectID != "" {
+		pyxisClient := pyxis.NewPyxisClient(
+			cfg.PyxisHost,
+			cfg.PyxisAPIToken,
+			cfg.CertificationProjectID,
+			&http.Client{Timeout: 60 * time.Second},
+		)
+		certProject, err := pyxisClient.GetProject(ctx)
 		if err != nil {
+			log.Error(fmt.Errorf("%w: %s", ErrRetrievingProject, err))
 			return err
 		}
+		log.Debugf("Certification project name is: %s", certProject.Name)
+		if certProject.Container.OsContentType == "scratch" {
+			cfg.Policy = policy.PolicyScratch
+			cfg.Scratch = true
+		}
 
-		// create the results file early to catch cases where we are not
-		// able to write to the filesystem before we attempt to execute checks.
-		resultsFile, err := os.OpenFile(
-			filepath.Join(artifacts.Path(), resultsFilenameWithExtension(formatter.FileExtension())),
-			os.O_WRONLY|os.O_CREATE|os.O_TRUNC,
-			0o600,
-		)
-		if err != nil {
-			return err
+		// if a partner sets `Host Level Access` in connect to `Privileged`, enable RootExceptionContainerPolicy checks
+		if certProject.Container.Privileged {
+			cfg.Policy = policy.PolicyRoot
 		}
+	}
+	engine, err := engine.NewForConfig(cfg.ReadOnly())
+	if err != nil {
+		return err
+	}
 
-		// also write to stdout
-		resultsOutputTarget := io.MultiWriter(os.Stdout, resultsFile)
+	formatter, err := formatters.NewForConfig(cfg.ReadOnly())
+	if err != nil {
+		return err
+	}
 
-		// At this point, we would no longer want usage information printed out
-		// on error, so it doesn't contaminate the output.
-		cmd.SilenceUsage = true
+	// create the results file early to catch cases where we are not
+	// able to write to the filesystem before we attempt to execute checks.
+	resultsFile, err := os.OpenFile(
+		filepath.Join(artifacts.Path(), resultsFilenameWithExtension(formatter.FileExtension())),
+		os.O_WRONLY|os.O_CREATE|os.O_TRUNC,
+		0o600,
+	)
+	if err != nil {
+		return err
+	}
 
-		// execute the checks
-		if err := engine.ExecuteChecks(ctx); err != nil {
-			return err
-		}
-		results := engine.Results(ctx)
+	// also write to stdout
+	resultsOutputTarget := io.MultiWriter(os.Stdout, resultsFile)
 
-		// return results to the user and then close output files
-		formattedResults, err := formatter.Format(ctx, results)
-		if err != nil {
-			return err
-		}
+	// execute the checks
```

Comment:
Here is where the cmd.SilenceUsage was previously. If it is not here, it doesn't have the same functionality.

Reply:
Continuing this discussion in a previous thread.